receiver/receivercreator/factory.go

```diff
@@ -70,10 +74,17 @@ func customUnmarshaler(sourceViperSection *viper.Viper, intoCfg interface{}) err
 		return err
 	}
 
-	receiversCfg := viperSub(sourceViperSection, receiversConfigKey)
+	receiversCfg, err := config.ViperSubExact(sourceViperSection, receiversConfigKey)
+	if err != nil {
+		return fmt.Errorf("failed to create sub viper for key %q", receiversConfigKey)
+	}
 
 	for subreceiverKey := range receiversCfg.AllSettings() {
-		cfgSection := viperSub(receiversCfg, subreceiverKey).GetStringMap(configKey)
+		subViperCfg, err := config.ViperSubExact(receiversCfg, subreceiverKey)
+		cfgSection := subViperCfg.GetStringMap(configKey)
+		if err != nil {
```

Comment:
Should this error be returned before calling `GetStringMap`?

Reply:
Good catch, thanks!